### PR TITLE
(feat)helm: Allow dynamic dependency override (#2205)

### PR DESCRIPTION
**Pull request by FX-HAO**
Add the --requirements flag to install and upgrade commands, which overrides dependencies.

### CODE REVIEW

**Comment by bacongobbler** [comment]
Would you mind writing some usage documentation under `docs/`? For example, it might be useful to demonstrate how this feature can be best utilized as well as some example use cases under the [Using Helm](https://github.com/helm/helm/blob/master/docs/using_helm.md) section of the guide. Thanks!

**Review by bacongobbler — CHANGES_REQUESTED** [review]
awaiting docs before approval

**Comment by FX-HAO** [comment]
@bacongobbler Added docs

**Comment by bacongobbler** [comment]
I was kind of expecting something with a little more substance. Can you explain *why* someone would want to use this feature in the documentation rather than how to use it? :)

**Comment by bacongobbler** [comment]
For example, https://github.com/helm/helm/blob/master/docs/using_helm.md#customizing-the-chart-before-installing is a great example of how and why you'd want to use the `--values` flag. If I'm reading the docs I'd like to know and when the `--requirements` flag may be useful.

**Comment by FX-HAO** [comment]
@bacongobbler I have addressed your comments. Please let me know if anything else to be done.

File: cmd/helm/install.go
Comment (FX-HAO) [reply cmd/helm/install.go]: You can just make a copy of chart's `requirements.yaml` file and update the part that you want to change. With this way you can remove any dependency if `tags` or `condition` fields are absent.

**Comment by bacongobbler** [comment]
As per [the announcement on the mailing list](https://lists.cncf.io/g/cncf-helm/topic/last_v2_feature_release/34301238), Helm 2.15.0 is the last release where we will be merging new features for Helm 2. Because this PR will require a large amount of backwards compatibility testing ([and some cleanup](https://github.com/helm/helm/pull/4764/files#diff-259fe82e12a866f01123927480c7851b)), I'm going to close this as something we will not be able to get to in time for the 2.15 release.

If someone is interested in following up, please re-target a PR against the Helm 3 branch (currently the `dev-v3` branch, but it may soon become the `master` branch).

In the future, I'd also ask if you could please write more documentation on the use cases this PR is attempting to address (the "why"). The documentation in this PR describes how the feature works, but it does not describe why a user should use this flag, or how it can be used (like as you described in [this comment](https://github.com/helm/helm/pull/4764#discussion_r239306562)).

Thanks.

**Pull request closed without merging**
